python/lsst/pipe/tasks/extended_psf.py

```diff
@@ -0,0 +1,256 @@
+        self.focal_plane_regions[region_name] = FocalPlaneRegionExtendedPsf(
+            extended_psf_image=extended_psf_image, detector_list=detector_list)
+        for det in detector_list:
+            self.detectors_focal_plane_regions[det] = region_name
+
+    def __call__(self, detector=None):
+        """Return the appropriate extended PSF.
+
+        If the instance contains no extended PSF defined over focal plane
+        regions, the default extended PSF will be returned regardless of
+        whether a detector ID was passed as argument.
+
+        Parameters
+        ----------
+        detector : `int`, optional
+            Detector ID. If focal plane region PSFs are defined, is used to
+            determine which model to return.
+
+        Returns
+        -------
+        extendedPsfImage : `lsst.afw.image.MaskedImageF`
+            The extended PSF model. If this instance contains extended PSFs
+            defined over focal plane regions, the extended PSF model for the
+            region that contains ``detector`` is returned. If not, the default
+            extended PSF is returned.
+        """
+        if detector is None:
+            if self.default_extended_psf is None:
+                raise ValueError("No default extended PSF available; please provide detector number.")
+            return self.default_extended_psf
+        elif not self.focal_plane_regions:
+            return self.default_extended_psf
+        return self.get_regional_extended_psf(detector=detector)
+
+    def __len__(self):
+        """Returns the number of extended PSF models present in the instance.
+
+        Note that if the instance contains both a default model and a set of
+        focal plane region models, the length of the instance will be the
+        number of regional models, plus one (the default). This is true even
+        in the case where the default model is one of the focal plane
+        region-specific models.
+        """
+        n_regions = len(self.focal_plane_regions)
+        if self.default_extended_psf is not None:
+            n_regions += 1
+        return n_regions
+
+    def get_regional_extended_psf(self, region_name=None, detector=None):
+        """Returns the extended PSF for a focal plane region.
+
+        The region can be identified either by name, or through a detector ID.
+
+        Parameters
+        ----------
+        region_name : `str` or `None`, optional
+            Name of the region for which the extended PSF should be retrieved.
+            Ignored if  ``detector`` is provided. Must be provided if
+            ``detector`` is None.
+        detector : `int` or `None`, optional
+            If provided, returns the extended PSF for the focal plane region
+            that includes this detector.
+
+        Raises
+        ------
+        ValueError
+            Raised if neither ``detector`` nor ``regionName`` are provided.
```

Comment:
Not at all, I like improving my English when I can! It's fun, because in French it is equally acceptable to use singular or plural for null quantities.